torch/csrc/jit/codegen/cuda/ir_graphviz.cpp

```diff
@@ -0,0 +1,451 @@
+             << "style=filled, fillcolor=azure];\n";
+}
+
+void IrGraphGenerator::printValue(const Val* val, const std::string& label) {
+  graph_def_ << "    " << getid(val) << " [label=\"" << label
+             << "\", shape=rect, color=green, fontsize=10];\n";
+}
+
+std::string IrGraphGenerator::generate() {
+  // IrGraphGenerator instances are not reusable
+  TORCH_CHECK(graph_def_.str().empty());
+  TORCH_CHECK(visited_.empty());
+
+  // record detail level
+  graph_def_ << "// detail level: ";
+  switch (detail_level_) {
+    case DetailLevel::ComputeOnly:
+      graph_def_ << "compute only\n";
+      break;
+    case DetailLevel::Basic:
+      graph_def_ << "minimal\n";
+      break;
+    case DetailLevel::Explicit:
+      graph_def_ << "explicit\n";
+      break;
+    case DetailLevel::Verbose:
+      graph_def_ << "verbose\n";
+      break;
+    default:
+      TORCH_CHECK(!"Unexpected detail level");
+  }
+
+  graph_def_ << "digraph fusion_ir {\n"
+             << "  node [shape=circle, color=gray];\n"
+             << "  edge [color=black];\n";
+
+  // Compute graph
+  generateComputeGraph();
+
+  // Schedule graph
+  if (detail_level_ > DetailLevel::ComputeOnly) {
+    generateScheduleGraph();
+  }
+
+  // All expressions & values
+  // (These are otherwise unreacheable (dead) nodes)
+  if (detail_level_ >= DetailLevel::Verbose) {
+    for (const auto* expr : fusion_->unordered_exprs()) {
+      handle(expr);
+    }
+    for (const auto* val : fusion_->vals()) {
+      handle(val);
+    }
+  }
+
+  // Finally, print all arc definitions
+  for (const auto& arc : arcs_) {
+    graph_def_ << "  " << arc << ";\n";
+  }
+
+  graph_def_ << "}\n";
+
+  // Make sure that all referenced nodes have been visited
+  for (const auto& kv : id_map_) {
+    TORCH_CHECK(visited(kv.first));
+  }
+
+  return graph_def_.str();
+}
+
+void IrGraphGenerator::generateComputeGraph() {
+  graph_def_ << "  subgraph cluster_compute {\n"
+             << "    label=\"compute\";\n"
+             << "    style=dashed;\n";
+
+  // Inputs
+  for (const auto* input : fusion_->inputs()) {
+    handle(input);
+  }
+
+  // Outputs
+  for (const auto* output : fusion_->outputs()) {
+    handle(output);
+  }
+
+  graph_def_ << "  }\n";
+}
+
+void IrGraphGenerator::generateScheduleGraph() {
+  graph_def_ << "  subgraph cluster_schedule {\n"
+             << "    label=\"schedule\";\n"
+             << "    style=dashed;\n";
+
+  // Connect TensorView with their TensorDomain
+  // (this will trigger the traversal of the schedule graph)
+  for (auto tv : tensor_views_) {
+    addArc(tv->domain(), tv, "[style=dashed, arrowhead=none]");
+    if (detail_level_ >= DetailLevel::Explicit) {
+      const auto root_domain = tv->getRootDomain();
+      addArc(tv, root_domain, "[style=dashed, color=green, arrowhead=none]");
+    }
+  }
+
+  graph_def_ << "  }\n";
+}
+
+void IrGraphGenerator::handle(const Statement* s) {
+  OptInConstDispatch::handle(s);
```

Comment:
If we are not really overriding the `handle(const Statement* s)` from inherited class, maybe we don't want to have a dummy implementation here.
You should be able to get it working with explicit `using OptInConstDispatch::handle` in the class declaration.